cli/src/pcluster/config/cluster_config.py

```diff
@@ -438,16 +436,29 @@ def _register_validators(self):
         self._register_validator(PlacementGroupIdValidator, placement_group_id=self.id)
 
 
-class QueueNetworking(_BaseNetworking):
+class _QueueNetworking(_BaseNetworking):
     """Represent the networking configuration for the Queue."""
 
-    def __init__(
-        self, subnet_ids: List[str], placement_group: PlacementGroup = None, assign_public_ip: str = None, **kwargs
-    ):
+    def __init__(self, subnet_ids: List[str], assign_public_ip: str = None, **kwargs):
         super().__init__(**kwargs)
         self.assign_public_ip = Resource.init_param(assign_public_ip)
         self.subnet_ids = Resource.init_param(subnet_ids)
+
+
+class SlurmQueueNetworking(_QueueNetworking):
+    """Represent the networking configuration for the slurm Queue."""
+
+    def __init__(self, subnet_ids: List[str], placement_group: PlacementGroup = None, proxy: Proxy = None, **kwargs):
+        super().__init__(subnet_ids, **kwargs)
         self.placement_group = placement_group
+        self.proxy = proxy
+
+
+class AwsBatchQueueNetworking(_QueueNetworking):
```

Comment:
Why do we need AwsBatchQueueNetworking? Is it the same as _QueueNetworking?

Reply:
Although QueueNetworking has same properties as AwsBatchQueueNetworking, AwsBatchQueueNetworking and SlurmQueueNetworking Inherit from QueueNetworking, and we should not expose QueueNetworking to avoid confusion.

Reply:
Is there some thing we can do to prevent to directly create instance from  _QueueNetworking class or use it? For example, define it as abstract class?

Reply:
I don't think we can set it as abstract class, otherwise _register_validators can't be used in parent class BaseNetworking. Underscore prefix helps us to distinguish it with other exposed classes.